kernel/nativevalues.ml

```diff
@@ -113,10 +113,17 @@ let mk_rel_accu i =
   mk_accu (Arel i)
 
 let rel_tbl_size = 100
-let rel_tbl = Array.init rel_tbl_size mk_rel_accu
+let rel_tbl_init = ref false
+(* Initialize the table lazily not to generate naked pointers at startup *)
+let rel_tbl = ref [||]
```

Comment:
You do not need a reference. Just initialize the array with a dummy value, e.g., `mk_int 0`.

Reply:
Nice, this was a problem in the multicore branch, I dunno tho if it wouldn't be better just to wrap the value in a `Lazy.t` ?

Reply:
I fear even the `mk_int 0` could be a problem in the future.

Reply:
If `mk_int` becomes a problem in the future, you can basically dump the native compiler in a trashbin.

Reply:
> I fear even the `mk_int 0` could be a problem in the future.

That does not make sense to me. If the OCaml runtime is changed to force boxing integers on the heap, the language is just doomed.

Reply:
Well indeed my comment goes beyond the scope of this PR, and it has to do with what happens with native and the multicore GC.